### PR TITLE
Implemented PUT route for skills and added corresponding test

### PR DESCRIPTION
**Pull request by nfallah**
This pull request closes issue #6. It allows users to edit specific skills via the 'id' query parameter; values outside the valid range will return a generic error, as will inputs that do not conform to the JSON format for a skill. A corresponding test has been added to show that the returned JSON indeed matches the user-requested JSON.

### CODE REVIEW

**Comment by nfallah** [comment]
@Pradyuman7

File: app.py [hunk 1]
```diff
@@ -307,6 +307,23 @@ def skill():
             201,
         )
 
+    if request.method == "PUT":
+        try:
+            skill_id = request.args.get('id')
+            if skill_id is None:
+                return jsonify({'error': 'Skill ID required'}), 400
```
Comment (Pradyuman7) [inline app.py:314]: nit: Maybe 'skill not found' instead?

File: app.py [hunk 2]
```diff
@@ -307,6 +307,23 @@ def skill():
...
+            skill_id = int(skill_id)
+            if skill_id < 0 or skill_id >= len(data['skill']):
+                raise IndexError
+            error = validate_fields(['name', 'proficiency', 'logo'], request.json)
+            if error:
+                return jsonify({"error": ", ".join(error) + " parameter(s) is required"}), 400
+            data['skill'][skill_id] = request.json
+            return jsonify(data['skill'][skill_id]), 200
+        except ValueError:
+            return jsonify({'error': 'Invalid skill ID'}), 400
```
Comment (Pradyuman7) [inline app.py:324]: Same here

**Comment by nfallah** [comment]
@Pradyuman7 I have updated the error messages and also attempted to fix any incoming merge conflicts. Does everything look good?

**Review by Pradyuman7 — APPROVED** [review]
LGTM 💪

**Pull request merged**